File: app/src/main/java/com/juniperphoton/myersplash/cloudservice/CategoryService.java
package com.juniperphoton.myersplash.cloudservice;

import com.juniperphoton.myersplash.model.UnsplashCategory;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface CategoryService {
    @GET("categories")
    Observable<List<UnsplashCategory>> getCategories(@Query("client_id") String id);
}

File: app/src/main/kotlin/com/juniperphoton/myersplash/cloudservice/CategoryService.kt
package com.juniperphoton.myersplash.cloudservice

import com.juniperphoton.myersplash.model.UnsplashCategory

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface CategoryService {
    @GET("categories")
    fun getCategories(@Query("client_id") id: String): Observable<MutableList<UnsplashCategory?>?>
}
